app/src/test/kotlin/no/nav/tiltakspenger/vedtak/service/BehandlingServiceTest.kt

```diff
@@ -258,4 +271,37 @@ internal class BehandlingServiceTest {
         lagretBehandling.captured.tiltak.first { it.id == "slutterInni" }.id shouldBe "slutterInni"
         lagretBehandling.captured.tiltak.first { it.id == "starterInni" }.id shouldBe "starterInni"
     }
+
+    @Test
+    fun `sjekk at man ikke kan se behandlinger for en person som er fortrolig uten tilgang`() {
+        val person = listOf(ObjectMother.personopplysningKjedeligFyr(fortrolig = true))
+        val behandlinger: List<Behandling> = ObjectMother.sakMedOpprettetBehandling(
+            personopplysninger = SakPersonopplysninger(person),
+        ).behandlinger
+
+        every { behandlingRepo.hentAlleForIdent(any()) } returns listOf(behandlinger.first() as Søknadsbehandling)
+        every { personopplysningService.hent(any()) } returns SakPersonopplysninger(person)
+
+        behandlingService.hentBehandlingForIdent("whatever", saksbehandler123()).size shouldBe 0
+        behandlingService.hentBehandlingForIdent("whatever", saksbehandlerMedKode7()).size shouldBe 1
+        // TODO : Er ikke dette galt?
+        behandlingService.hentBehandlingForIdent("whatever", saksbehandlerMedKode6()).size shouldBe 0
+    }
+
+    @Test
+    @Disabled("Fungerer ikke fordi vi har transaksjonskode i BehandlingServiceImpl")
+    fun `sjekk at man ikke kan sende inn uten beslutter rolle`() {
```

Comment:
Dette synes jeg er en viktig test å ha, er det noen måte å løse problematikken med transaksjonskode for å kunne enable testen igjen?

Reply:
Ja, men det får bli en egen PR tror jeg.

Reply:
Jeg har en plan der også! :)

Reply:
Supert! Gleder meg til å se 😊